scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -455,3 +456,63 @@ def test_from_euler_extrinsic_rotation_313():
         [0.91855865, -0.30618622, -0.250000],
         [0.35355339, 0.35355339, 0.8660254]
     ]))
+
+
+def test_as_euler_extrinsic_rotation():
```

Comment:
I would probably split tests for symmetric and asymmetric axis sequence (i.e. zxz and xyx types). And test for intrinsic and extrinsic rotations at the same place.

For zyx type permutations are correct approach, but for zxz perhaps just use two nested loops.

My point is that the code could be a little bit more readable.